test/tests/kernels/body_force/ad_mat_forcing_function_test.i

```diff
@@ -0,0 +1,62 @@
+[Mesh]
+  [square]
+    type = GeneratedMeshGenerator
+    nx = 2
+    ny = 2
+    dim = 2
+  []
+  uniform_refine = 4
+[]
+
+[Variables]
+  [u]
+  []
+[]
+
+[Materials]
+  [forcing_material]
+    type = ADParsedMaterial
+    property_name = forcing_material
+    extra_symbols = x
+    expression = 'alpha*alpha*pi*pi*sin(alpha*pi*x)'
+    constant_names = 'alpha pi'
+    constant_expressions = '16 3.14159265359'
+  []
+[]
+
+[Kernels]
+  [diff]
+    type = ADDiffusion
+    variable = u
+  []
+  [forcing]
+    type = ADMatBodyForce
```

Comment:
Probably can delete this input file and just pass `ADParsedMaterial` and `ADMatBodyForce` via cliargs